### ISSUE TITLE
Add validation for WAF/mod_security configuration 

### ISSUE THREAD

**Issue by bdecoste** [solo-io/gloo#8139]
### Version

None

### Is your feature request related to a problem? Please describe.

Gloo does not validate the WAF/mod_security configuration so we can run into a situation where Envoy rejects configuration and we can only detect the rejected config by exampling the Envoy logs. For example:

`gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: Rules error. File: /modsecurity/rules/crs/REQUEST-901-INITIALIZATION.conf. Line: 53. Column: 40. Invalid input:  @eq 0" \`

caused by a missing new line at the end of the waf part of the gloo config

```
waf:
    ruleSets:
      - ruleStr: |
          SecRuleUpdateTargetByTag "OWASP_CRS/WEB_ATTACK/SQL_INJECTION" "!REQUEST_COOKIES"%
```

### Describe the solution you'd like

Have gloo validate the WAF/mod_security config so that invalid config is not sent to Envoy but rather rejected at the control plane or validating webhook 

### Describe alternatives you've considered

None

**Comment by SantoDE** [comment]
It will rejects ongoing updates for a given route.

**Comment by DuncanDoyle** [comment]
Fixed in 1.17.0.
